servers/src/main/java/tachyon/worker/block/TieredBlockStore.java

```diff
@@ -441,31 +473,41 @@ private void freeSpaceInternal(long userId, long availableBytes, BlockStoreLocat
     for (int alias : destTierAlias) {
       Set<Pair<Long, BlockStoreLocation>> toMove = blocksGroupedByDestTier.get(alias);
       for (Pair<Long, BlockStoreLocation> entry : toMove) {
-        long blockId = entry.getFirst();
-        BlockStoreLocation newLocation = entry.getSecond();
-        BlockMeta blockMeta = mMetaManager.getBlockMeta(blockId);
-        BlockStoreLocation oldLocation = blockMeta.getBlockLocation();
-        long lockId = mLockManager.lockBlock(userId, blockId, BlockLockType.WRITE);
         try {
-          moveBlockNoLock(blockId, newLocation);
-          synchronized (mBlockStoreEventListeners) {
-            for (BlockStoreEventListener listener : mBlockStoreEventListeners) {
-              listener.onMoveBlockByWorker(userId, blockId, oldLocation, newLocation);
+          long blockId = entry.getFirst();
+          BlockStoreLocation newLocation = entry.getSecond();
+          BlockMeta blockMeta = mMetaManager.getBlockMeta(blockId);
+          BlockStoreLocation oldLocation = blockMeta.getBlockLocation();
+          long lockId = mLockManager.lockBlock(userId, blockId, BlockLockType.WRITE);
+          try {
+            moveBlockNoLock(blockId, newLocation);
+            synchronized (mBlockStoreEventListeners) {
+              for (BlockStoreEventListener listener : mBlockStoreEventListeners) {
+                listener.onMoveBlockByWorker(userId, blockId, oldLocation, newLocation);
+              }
             }
+          } catch (IOException ioe) {
+            throw new OutOfSpaceException("Failed to free space: cannot move block " + blockId
+                + " to " + newLocation, ioe);
+          } catch (AlreadyExistsException aee) {
+            throw new OutOfSpaceException("Failed to free space: block " + blockId
+                + " already exists in " + newLocation, aee);
+          } catch (InvalidStateException fpe) {
+            throw new OutOfSpaceException("Failed to free space: block " + blockId
+                + " is still temporary", fpe);
```

Comment:
+1, I feel it is tricky to simply throw all `OutOfSpaceException` here rather than their original type of Exceptions. I guess it is probably inevitable that we need to do some Exception conversion somewhere, but I feel it shouldn't be here